pkg/pipeline/lifecycle_test.go

```diff
@@ -414,34 +435,23 @@ func generateRecords(count int) []record.Record {
 func generatorSource(
 	ctrl *gomock.Controller,
 	persister *connector.Persister,
-	records []record.Record,
+	records []opencdc.Record,
 	wantErr error,
-	teardown bool,
+	stop bool,
 ) (*connector.Instance, *pmock.Dispenser) {
-	position := 0
-	recordCount := len(records)
-
-	sourcePlugin := pmock.NewSourcePlugin(ctrl)
-	sourcePlugin.EXPECT().Start(gomock.Any(), nil).Return(nil).Times(1)
-	sourcePlugin.EXPECT().Configure(gomock.Any(), gomock.Any()).Return(nil).Times(1)
-
-	if teardown {
-		sourcePlugin.EXPECT().Teardown(gomock.Any()).Return(nil).Times(1)
+	sourcePluginOptions := []pmock.ConfigurableSourcePluginOption{
+		pmock.SourcePluginWithConfigure(),
+		pmock.SourcePluginWithOpen(),
+		pmock.SourcePluginWithRun(),
+		pmock.SourcePluginWithRecords(records, wantErr),
+		pmock.SourcePluginWithAcks(len(records), wantErr == nil),
+		pmock.SourcePluginWithTeardown(),
 	}
 
-	sourcePlugin.EXPECT().Ack(gomock.Any(), gomock.Any()).Return(nil).Times(recordCount)
-	sourcePlugin.EXPECT().Read(gomock.Any()).DoAndReturn(func(ctx context.Context) (record.Record, error) {
-		if position == recordCount {
-			if wantErr != nil {
-				return record.Record{}, wantErr
-			}
-			<-ctx.Done()
-			return record.Record{}, ctx.Err()
-		}
-		r := records[position]
-		position++
-		return r, nil
-	}).MinTimes(recordCount + 1)
+	if stop {
+		sourcePluginOptions = append(sourcePluginOptions, pmock.SourcePluginWithStop())
+	}
+	sourcePlugin := pmock.NewConfigurableSourcePlugin(ctrl, sourcePluginOptions...)
```

Comment:
This is where we use `ConfigurableSourcePlugin` and `ConfigurableDestinationPlugin` mentioned in the other comment.